src/tests/unittests/test_data.py

```diff
@@ -259,27 +332,73 @@ def test_makedataset1d_shape_measuredata_list_nok2(self):
+            stream_handler = logging.StreamHandler(sys.stdout)
+            logging.getLogger().addHandler(stream_handler)
+
+            data_set = qtt.data.makeDataSet1D(x, yname, y, return_names=False)
+            self.assertTrue(np.array_equal(data_set.dummy, x))
+            self.assertTrue(np.array_equal(data_set.measured, y))
+
+            # Verify warning
+            print_string = mock_stdout.getvalue()
+            self.assertIn('Shape of measured data does not match setpoint shape', print_string)
+            logging.getLogger().removeHandler(stream_handler)
 
     def test_makedataset1d_shape_measuredata_nok(self):
         p = ManualParameter('dummy')
         x = p[0:10:1]
         y = [np.arange(len(x)+1)]
+        with patch('sys.stdout', new_callable=io.StringIO) as mock_stdout:
+            stream_handler = logging.StreamHandler(sys.stdout)
+            logging.getLogger().addHandler(stream_handler)
+
+            data_set = qtt.data.makeDataSet1D(x, 'y', y, return_names=False)
+            self.assertTrue(np.array_equal(data_set.dummy, x))
+            self.assertTrue(np.array_equal(data_set.y, y))
 
-        self.assertRaisesRegex(ValueError,
-                               'Measured data must be a sequence with shape matching the setpoint arrays shape',
-                               qtt.data.makeDataSet1D, x, 'y', y, return_names=False)
+            # Verify warning
+            print_string = mock_stdout.getvalue()
+            self.assertIn('Shape of measured data does not match setpoint shape', print_string)
+            logging.getLogger().removeHandler(stream_handler)
 
-    def test_makedataset1d_shape_measuredata_second_nok(self):
+    def test_makedataset1d_shape_second_item_measuredata_nok(self):
         p = ManualParameter('dummy')
         x = p[0:10:1]
         y = [np.arange(len(x)), np.arange(len(x)+1)]
+        with patch('sys.stdout', new_callable=io.StringIO) as mock_stdout:
+            stream_handler = logging.StreamHandler(sys.stdout)
+            logging.getLogger().addHandler(stream_handler)
+
+            data_set = qtt.data.makeDataSet1D(x, ['y1', 'y2'], y, return_names=False)
+            self.assertTrue(np.array_equal(data_set.dummy, x))
+            self.assertTrue(np.array_equal(data_set.y1, y[0]))
+            self.assertTrue(np.array_equal(data_set.y2, y[1]))
 
-        self.assertRaisesRegex(ValueError,
-                               'Measured data must be a sequence with shape matching the setpoint arrays shape',
-                               qtt.data.makeDataSet1D, x,
-                               ['y1', 'y2'], y, return_names=False)
+            # Verify warning
+            print_string = mock_stdout.getvalue()
+            self.assertIn('Shape of measured data does not match setpoint shape', print_string)
+            logging.getLogger().removeHandler(stream_handler)
+
+    def test_makedataset1d_no_data(self):
+        p = ManualParameter('dummy')
```

Comment:
The `p`'s could be better named into something more meaningful like e.g. `dummy_parameter`?

Reply:
renamed